plugins/notifications-backend-module-email-notification-processor/src/module.ts

```diff
@@ -0,0 +1,175 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  coreServices,
+  createBackendModule,
+  LoggerService,
+} from '@backstage/backend-plugin-api';
+import { CatalogApi, CatalogClient } from '@backstage/catalog-client';
+import {
+  NotificationSeverity,
+  Notification,
+} from '@backstage/plugin-notifications-common';
+import {
+  notificationsProcessingExtensionPoint,
+  NotificationProcessor,
+} from '@backstage/plugin-notifications-node';
+import { JsonObject } from '@backstage/types';
+import { createTransport } from 'nodemailer';
+
+function severityToNumber(severity: NotificationSeverity | undefined): number {
+  switch (severity) {
+    case undefined:
+      return Number.MIN_VALUE;
+    case 'critical':
+      return 4;
+    case 'high':
+      return 3;
+    case 'normal':
+      return 2;
+    case 'low':
+      return 1;
+    default:
+      throw new Error(`notification severity has an unknown value ${severity}`);
+  }
+}
+
+class EmailNotificationProcessor implements NotificationProcessor {
+  threshold: number;
+  sender: string;
+  broadcastRecipients: string[];
+  transport: any;
+  catalogApi: CatalogApi;
+  logger: LoggerService;
+
+  constructor(
+    catalogApi: CatalogApi,
+    logger: LoggerService,
+    threshold: number,
+    sender: string,
+    broadcastRecipients: string[],
+    hostname: string,
+    port: number,
+    secure?: boolean,
+    requireTLS?: boolean,
+    username?: string,
+    password?: string,
+  ) {
+    this.threshold = threshold;
+    this.sender = sender;
+    this.broadcastRecipients = broadcastRecipients;
+    this.transport = createTransport({
+      host: hostname,
+      port: port,
+      secure: secure,
+      requireTLS: requireTLS,
+      auth: {
+        user: username,
+        pass: password,
+      },
+      logger: true,
+    });
+    this.catalogApi = catalogApi;
+    this.logger = logger;
+  }
+  getName(): string {
+    return 'Email';
+  }
+
+  async send(notification: Notification): Promise<void> {
+    const numericSeverity = severityToNumber(notification.payload.severity);
+
+    if (numericSeverity < this.threshold) {
+      return;
+    }
+
+    let recipient;
+
+    if (notification.user) {
+      await this.catalogApi.getEntityByRef(notification.user).then(entity => {
+        recipient = (entity?.spec?.profile as JsonObject)?.email;
+      });
+
+      if (!recipient) {
+        this.logger.warn(`notification ${notification.id} has no recipient`);
+        return;
+      }
+    } else {
+      recipient = this.broadcastRecipients;
```

Comment:
Why not just fetch all users from the catalog for this?

Reply:
all users? I don't think it is a good idea. We had a discussion about it including Marek and this is what was decided. @mareklibra.
Besides, you don't have any processors yet and you don't have any other method of sending notifications. So what's wrong with starting with this?

Reply:
There's the slack processor PR that's looking very good already

Reply:
in the slack processor you send to all users?

Reply:
do u provide an api for querying such a large amount of users?

Reply:
I would make either the list of email recipients configurable or provide configuration to filter. This can be very dangerous functionality, easy to flood the users or do DDoS, mind there can be tens of thousands users, even inactive ones.

IMO, having such configurable list of email addresses to receive broadcasts is a good way to go. It will be up to the external SMTP server to forward the emails farther.